src/storage/src/hummock/compactor/compactor_runner.rs

```diff
@@ -226,29 +223,57 @@ mod tests {
                 1,
             ),
         ];
-        let sstable_info = gen_test_sstable_with_range_tombstone(
+        let mut sstable_info_1 = gen_test_sstable_with_range_tombstone(
             default_builder_opt_for_test(),
             1,
+            kv_pairs.clone().into_iter(),
+            range_tombstones.clone(),
+            sstable_store.clone(),
+        )
+        .await
+        .get_sstable_info();
+        sstable_info_1.table_ids = vec![1];
+
+        let mut sstable_info_2 = gen_test_sstable_with_range_tombstone(
+            default_builder_opt_for_test(),
+            2,
             kv_pairs.into_iter(),
             range_tombstones.clone(),
             sstable_store.clone(),
         )
         .await
         .get_sstable_info();
+        sstable_info_2.table_ids = vec![2];
+
         let compact_task = CompactTask {
             input_ssts: vec![InputLevel {
                 level_idx: 0,
                 level_type: 0,
-                table_infos: vec![sstable_info],
+                table_infos: vec![sstable_info_1, sstable_info_2],
             }],
             existing_table_ids: vec![2],
             ..Default::default()
         };
         let mut state_clean_up_filter = StateCleanUpCompactionFilter::new(HashSet::from_iter(
             compact_task.existing_table_ids.clone(),
         ));
+
+        let sstable_infos = compact_task
+            .input_ssts
+            .iter()
+            .flat_map(|level| level.table_infos.iter())
+            .filter(|table_info| {
+                let table_ids = &table_info.table_ids;
+                table_ids
+                    .iter()
+                    .any(|table_id| compact_task.existing_table_ids.contains(table_id))
```

Comment:
What about sort `existing_table_ids` and use binary search here?

Reply:
fix it